lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,463 @@
+	 *
+	 * @return bool True if the font family has font faces defined, false otherwise.
+	 */
+	public function has_font_faces() {
+		return ! empty( $this->data['fontFace'] ) && is_array( $this->data['fontFace'] );
+	}
+
+	/**
+	 * Removes font family assets.
+	 *
+	 * @return bool True if assets were removed, false otherwise.
+	 */
+	private function remove_font_family_assets() {
+		if ( $this->has_font_faces() ) {
+			foreach ( $this->data['fontFace'] as $font_face ) {
+				$were_assets_removed = $this->delete_font_face_assets( $font_face );
+				if ( false === $were_assets_removed ) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Removes a font family from the database and deletes its assets.
+	 *
+	 * @return bool|WP_Error True if the font family was uninstalled, WP_Error otherwise.
+	 */
+	public function uninstall() {
+		$post = $this->get_data_from_post();
+		if ( null === $post ) {
+			return new WP_Error( 'font_family_not_found', __( 'The font family could not be found.', 'gutenberg' ) );
+		}
+		if ( ! $this->remove_font_family_assets() || ! wp_delete_post( $post->ID, true ) ) {
+			return new WP_Error( 'font_family_not_deleted', __( 'The font family could not be deleted.', 'gutenberg' ) );
+		}
+		return true;
+	}
+
+	/**
+	 * Deletes a specified font asset file from the fonts directory.
+	 *
+	 * @param string $src The path of the font asset file to delete.
+	 * @return bool Whether the file was deleted.
+	 */
+	private static function delete_asset( $src ) {
+		$filename  = basename( $src );
+		$file_path = path_join( WP_Fonts_Library::get_fonts_dir(), $filename );
+		wp_delete_file( $file_path );
+		return ! file_exists( $file_path );
+	}
+
+	/**
+	 * Deletes all font face asset files associated with a given font face.
+	 *
+	 * @param array $font_face The font face array containing the 'src' attribute with the file path(s) to be deleted.
+	 */
+	private static function delete_font_face_assets( $font_face ) {
+		$sources = (array) $font_face['src'];
+		foreach ( $sources as $src ) {
+			$was_asset_removed = self::delete_asset( $src );
+			if ( ! $was_asset_removed ) {
+				// Bail if any of the assets could not be removed.
+				return false;
+			}
+		}
+		return true;
+	}
+
+
+	/**
+	 * Gets the overrides for the 'wp_handle_upload' function.
+	 *
+	 * @param string $filename The filename to be used for the uploaded file.
+	 *
+	 * @return array The overrides for the 'wp_handle_upload' function.
+	 */
+	private function get_upload_overrides( $filename ) {
+		return array(
+
+			/*
+			 * Arbitrary string to avoid the is_uploaded_file() check applied
+			 * when using 'wp_handle_upload'.
+			 */
+			'action'                   => 'wp_handle_font_upload',
+			// We are not testing a form submission.
+			'test_form'                => false,
+
+			/*
+			 * Seems like we can not test mime type for files that are not images.
+			 * See wp_check_filetype_and_ext().
+			 */
+			'test_type'                => false,
```

Comment:
[_wp_handle_upload()](https://developer.wordpress.org/reference/functions/_wp_handle_upload/) documents a `'mimes'` key for `$overrides` which gets passed to `wp_check_filetype_and_ext()`, so would this something like this work?

```php
'test_type' => true,
'mimes'     => array(
	'woff' => PHP_VERSION_ID >= 80112 ? 'font/woff' : 'application/font-woff',
	'and'  => 'so on',
);
```

Reply:
Hmm interesting. What do you think @matiasbenedetto?

Reply:
I tried that before, and it didn't work as expected. Not completely sure why. I'll try again to confirm.

Reply:
Added a follow up issue for this: https://github.com/WordPress/gutenberg/issues/53576